client/state/data-layer/wpcom/user-settings/index.js

```diff
@@ -0,0 +1,83 @@
+/**
+ * External dependencies
+ */
+import { isEmpty, keys, merge, noop } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import { decodeEntities } from 'lib/formatting';
+import { http } from 'state/data-layer/wpcom-http/actions';
+import { dispatchRequest } from 'state/data-layer/wpcom-http/utils';
+import { updateUserSettings, clearUnsavedUserSettings } from 'state/user-settings/actions';
+import {
+	USER_SETTINGS_REQUEST,
+	USER_SETTINGS_SAVE,
+} from 'state/action-types';
+
+/*
+ * Decodes entities in those specific user settings properties
+ * that the REST API returns already HTML-encoded
+ */
+function fromApi( apiResponse ) {
+	const decodedValues = {
+		display_name: apiResponse.display_name && decodeEntities( apiResponse.display_name ),
+		description: apiResponse.description && decodeEntities( apiResponse.description ),
+		user_URL: apiResponse.user_URL && decodeEntities( apiResponse.user_URL )
+	};
+
+	return merge( {}, apiResponse, decodedValues );
+}
+
+export const requestUserSettings = ( { dispatch }, action, next ) => {
+	dispatch( http( {
+		apiVersion: '1.1',
+		method: 'GET',
+		path: '/me/settings',
+	}, action ) );
+
+	return next( action );
+};
+
+export const receiveUserSettings = ( { dispatch }, action, next, data ) => {
+	dispatch( updateUserSettings( fromApi( data ) ) );
+};
+
+export const receiveRequestError = noop;
+
+/*
+ * Post settings to WordPress.com API at /me/settings endpoint
+ */
+export function saveUserSettings( { dispatch, getState }, action, next ) {
+	const { settingsOverride } = action;
+	const settings = settingsOverride || getState().userSettings.unsavedSettings;
+
+	if ( ! isEmpty( settings ) ) {
+		dispatch( http( {
+			apiVersion: '1.1',
+			method: 'POST',
+			path: '/me/settings',
+			body: settings,
+		}, action ) );
+	}
+
+	return next( action );
+}
+
+export const receiveSavedUserSettings = ( { dispatch }, { settingsOverride }, next, data ) => {
```

Comment:
the name here I think could be more expressive in what it's doing. it's maybe not so much _receiving_ as it is _updating_ or _storing_. if we called this function without an API call being attached the nuance might look clearer:

```js
// in dev tools
receiveSavedUserSettings( store, settingsAction, store.dispatch, {
	… // mock settings data
} );
```

this wasn't _received_ any more than other direct function calls are _parameter receivers_